### ISSUE TITLE
Momentum based gradient descent

### ISSUE THREAD

**Issue by wdrap** [wdrap/proxima-js#1]
Implement gradient descent with momentum. It should be added as hyperparameter.
